beacon-chain/powchain/service.go

```diff
@@ -324,6 +328,37 @@ func (s *Service) IsConnectedToETH1() bool {
 	return s.connectedETH1
 }
 
+// CurrentETH1Endpoint returns the URL of the current ETH1 endpoint.
+func (s *Service) CurrentETH1Endpoint() string {
+	return s.cfg.currHttpEndpoint.Url
+}
+
+// CurrentETH1ConnectionError returns the error (if any) of the current connection.
+func (s *Service) CurrentETH1ConnectionError() error {
+	_, _, err := s.dialETH1Nodes(s.cfg.currHttpEndpoint)
```

Comment:
The clients should be closed after we dial them otherwise there is a resource leak and the connection is
kept active.

Reply:
sorry, totally missed this. thanks for addressing!